dracut/modules.d/90kiwi-dump/kiwi-ramdisk-deployment-generator.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+type getarg >/dev/null 2>&1 || . /lib/dracut-lib.sh
+
+if ! getargbool 0 rd.kiwi.ramdisk; then
+    # we create a sysroot generator only for the special
+    # ramdisk deployment case
+    exit 0
+fi
+
+root_cmdline=$(getarg root=)
+if [ -z "${root_cmdline}" ];then
+    # if the root= parameter is missing there is no guarantee
+    # that the /config.bootoptions file is present on local
+    # storage at the time this generator code is called. This
+    # case applies on PXE deployments which serves config.bootoptions
+    # from the network. The mount process including config.bootoptions
+    # paramters is then handled by the dracut
+    # mount hook: kiwi-mount-ramdisk.sh
+    exit 0
+fi
+
+GENERATOR_DIR="$1"
+[ -z "${GENERATOR_DIR}" ] && exit 1
+[ -d "${GENERATOR_DIR}" ] || mkdir -p "${GENERATOR_DIR}"
+
+[ -e /config.bootoptions ] || exit 1
+
+root_uuid=$(
+    while read -r -d ' ' opt; do echo "${opt}";done < /config.bootoptions |\
+    grep root= | cut -f2- -d=
+)
+
+[ -z "${root_uuid}" ] && exit 1
+
+{
+    echo "[Unit]"
+    echo "Before=initrd-root-fs.target"
```

Comment:
I'd say we are missing the `DefaultDependencies=no` here. See this systemd issue here https://github.com/systemd/systemd/issues/12231 which was solved by adding 

```c
/* All mounts under /sysroot need to happen later, at initrd-fs.target time. IOW, it's not
 * technically part of the basic initrd filesystem itself, and so shouldn't inherit the default
 * Before=local-fs.target dependency. */                                
if (in_initrd() && path_startswith(where, "/sysroot"))                  
    fprintf(f, "DefaultDependencies=no\n");
```
into the fstab generator (this generator creates `sysroot.mount`).

We also fixed that a while ago in live generator here 4238b7eac9, but I guess we missed the ramdisk generator back at that time.

Reply:
yep good catch, this was never done in the live ramdisk generator. I'll adapt it

Reply:
done